code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,693 @@
+		examine_message += "\The [src] is full! "
+
+	examine_message += "<br>Cleanliness level: "
+
+	//Approximate filth level
+	if(filth_level == 0)
+		examine_message += "[src] is spotless! "
+	else if(filth_level <= 2.5)
+		examine_message += "[src] looks like the glass has been smudged. "
+	else if(filth_level <= 5)					//This is the breeding threshold
+		examine_message += "[src] has some algae growth in it. "
+	else if(filth_level <= 7.5)
+		examine_message += "[src] has a lot of algae growth in it. "
+	else if(filth_level < 10)
+		examine_message += "[src] is getting hard to see into! Someone should clean it soon! "
+	else if(filth_level == 10)
+		examine_message += "[src] is absolutely disgusting! Someone should clean it NOW! "
+
+	examine_message += "<br>Food level: "
+
+	//Approximate food level
+	if(!fish_count)								//Check if there are fish in the tank
+		if(food_level > 0)						//Don't report a tank that has neither fish nor food in it
+			examine_message += "There's some food in [src], but no fish! "
+	else										//We've got fish, report the food level
+		if(food_level == 0)
+			examine_message += "The fish look very hungry! "
+		else if(food_level < 2)
+			examine_message += "The fish are nibbling on the last of their food. "
+		else if(food_level < 10)				//Breeding is possible
+			examine_message += "The fish seem happy! "
+		else if(food_level == 10)
+			examine_message += "There is a solid layer of fish food at the top. "
+
+	//Report the number of harvestable eggs
+	if(egg_count)								//Don't bother if there isn't any eggs
+		examine_message += "<br>There are [egg_count] eggs able to be harvested! "
+
+	examine_message += "<br>"
+
+	//Report the number and types of live fish if there is water in the tank
+	if(fish_count == 0)
+		examine_message += "\The [src] doesn't contain any live fish. "
+	else
+		//Build a message reporting the types of fish
+		var/fish_num = fish_count
+		var/message = "You spot "
+		while(fish_num > 0)
+			if(fish_count > 1 && fish_num == 1)	//If there were at least 2 fish, and this is the last one, add "and" to the message
+				message += "and "
+			message += "a [fish_list[fish_num]]"
+			fish_num --
+			if(fish_num > 0)					//There's more fish, add a comma to the message
+				message +=", "
+		message +="."							//No more fish, end the message with a period
+		//Display the number of fish and previously constructed message
+		examine_message += "\The [src] contains [fish_count] live fish. [message] "
+
+	examine_message += "<br>"
+
+	//Report lid state for tanks and wall-tanks
+	if(has_lid)									//Only report if the tank actually has a lid
+		//Report lid state
+		if(lid_switch)
+			examine_message += "The lid is closed. "
+		else
+			examine_message += "The lid is open. "
+
+	examine_message += "<br>"
+
+	//Report if the tank is leaking/cracked
+	if(water_level > 0)							//Tank has water, so it's actually leaking
+		if(leaking == 1) examine_message += "\The [src] is leaking."
```

Comment:
@PJB3005 Specifically forbid me from one line ifs, so nobody else sure as hell gets to use them